base/doc/clsguide.tex

```diff
@@ -977,29 +978,46 @@ \subsection{Passing options around}
 handler, but this handler is not used for |landscape| because that option is
 explicitly declared.
 
-\subsection{Testing dates of other code}
+\subsection{Useful status tests}
+
+A number of utility status tests are available which can be used by package and
+class authors to query the presence and date of other code, the current kernel
+release and other related ideas. Three forms of each test are provided, one
+requiring arguments for both \meta{true} and \meta{false} branches, one
+requiring only a \meta{true} branch and one requiring only a \meta{false}
+branch. These are indicated in their names as |TF|, |T| and |F|, respectively.
+Here, we document only the |TF| versions: the other forms are however
+available.
+
+\begin{decl}
+  |\IfPackageLoadedTF| \arg{package-name} \arg{true code} \arg{false code}\\
+  |\IfClassLoadedTF| \arg{class-name} \arg{true code} \arg{false code}\\
+  |\IfFileLoadedTF| \arg{file-name} \arg{true code} \arg{false code}
+\end{decl}
+These commands test if the named package, class or file is loaded: this
+requires that a matching |\Provides...| line has been given, as described in
+Section~\ref{sec:ID}. In the case of \cs{IfFileLoadedTF}, the full
+\meta{file-name} must be provided; in contrast, no extension should be given if
+testing for a package or class.
 
 \begin{decl}
-  |\IfPackageAtLeast(TF)| \arg{package-name} \arg{date} \arg{true code} \arg{false code}\\
-  |\IfClassAtLeast(TF)| \arg{class-name} \arg{date} \arg{true code} \arg{false code}\\
-  |\IfFileAtLeast(TF)| \arg{file-name} \arg{date} \arg{true code} \arg{false code}
+  |\IfPackageAtLeastTF| \arg{package-name} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfClassAtLeastTF| \arg{class-name} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfFileAtLeastTF| \arg{file-name} \arg{date} \arg{true code} \arg{false code}
 \end{decl}
-There are occasions when you need to test if a package is
-at least a particular date (assuming it has been loaded). This can be done using the
-commands |\IfPackageAtLeastTF|, etc. Separate commands are available to test
-packages, classes and general files: in all cases, they rely on the existence
-of a |\Provides...| line in the file being queried. Three versions
-of each command are available, ending |T|, |F| and |TF|, which indicate the
-outcome branches required. Thus for example |\IfPakcageAtLeastT| does not need
-a \meta{false code} argument. The \meta{date} argument can be given either in
-ISO (\textsc{yyyy-mm-dd}) or traditional \LaTeX{} (\textsc{yyyy/mm/dd}) formats.
+These tests are used to check if a package or class has been loaded with at
+least a given \meta{date}. The \meta{date} is compared with that in the
+optional argument to \cs{Provides...} (as described in Section~\ref{sec:ID}).
+As for \cs{Provides...}, the \meta{date} should be given in the form
```

Comment:
I think somewhere probably at the referenced Provides.... section we need to say yyyy/mm/dd form can be used, as far as I can tell after this diff, all mention of that form is gone, but if anyone looks at any existing examples that's mostly what they will see,

Reply:
I'll adjust again - I'll add something in there (looking at the blame, I wrote the bit without the traditional date format).

(As an aside, I do wonder if we should normalise the dates across the repos ...)

Reply:
Perhaps: Both forms are accepted and the yyyy-mm-dd form is recommended in new codes.

Reply:
I'm wondering where that would fit: with `\ProvidesPackage`?

Reply:
well, in many cases I prefer the traditional /../../ notation

Reply:
Maybe this could be combined with a careful explanation of the important point that there must be exactly 8 digits used (4+2+2) for the date (using either / or -).

Reply:
I've added a note